tls/s2n_handshake_io.c

```diff
@@ -510,16 +515,21 @@ int s2n_generate_new_client_session_id(struct s2n_connection *conn)
 /* Lets the server flag whether a HelloRetryRequest is needed while processing extensions */
 int s2n_set_hello_retry_required(struct s2n_connection *conn)
 {
-    conn->handshake.handshake_type |= HELLO_RETRY_REQUEST;
+    bool has_correct_server_version = conn->server_protocol_version >= S2N_TLS13;
+    bool has_correct_client_version = conn->client_protocol_version == S2N_TLS13;
```

Comment:
\>=, not ==. Also, can you just return early when checking for these instead of storing their values and using them later? And should it be an error instead of just a no-op?

Reply:
We could error here.